### ISSUE TITLE
docker: broken since systemd 231->232

### ISSUE THREAD

**Issue by volth** [NixOS/nixpkgs#22472]
## Issue description

Docker does not work with systemd 232
See yet unfixed upstream issues https://github.com/docker/docker/issues/28109 https://github.com/opencontainers/runc/issues/1175

### Steps to reproduce

```
# docker run --rm hello-world                                                                                                                                                           
container_linux.go:247: starting container process caused "process_linux.go:359: container init caused \"rootfs_linux.go:54: mounting \\\"cgroup\\\" to rootfs \\\"/var/lib/docker/devicemapper/mnt
/658357677f4a7b445fa855df2e3d4212f0f1312c5cf55e6aa0247e8db8a9c026/rootfs\\\" at \\\"/sys/fs/cgroup\\\" caused \\\"no subsystem for mount\\\"\""                                                    
/nix/store/zy18pnk4wi1vynp3r83mk3a9zj3l069y-docker-1.13.0/libexec/docker/docker: Error response from daemon: oci runtime error: container_linux.go:247: starting container process caused "process_
linux.go:359: container init caused \"rootfs_linux.go:54: mounting \\\"cgroup\\\" to rootfs \\\"/var/lib/docker/devicemapper/mnt/658357677f4a7b445fa855df2e3d4212f0f1312c5cf55e6aa0247e8db8a9c026/r
ootfs\\\" at \\\"/sys/fs/cgroup\\\" caused \\\"no subsystem for mount\\\"\"".                                                                                                                      
```                               

## Technical details

* System: (NixOS: `nixos-version`, Ubuntu/Fedora: `lsb_release -a`, ...)
17.03.git.2794806 (Gorilla)

* Nix version: (run `nix-env --version`)
nix-env (Nix) 1.11.6

* Nixpkgs version: (run `nix-instantiate --eval '<nixpkgs>' -A lib.nixpkgsVersion`)
"17.03.git.27948069ec"

**Comment by grahamc** [comment]
@globin this is pretty bad. Thought you should see it, specifically. Thank you @volth for reporting!

**Comment by globin** [comment]
systemd 🙄 I'll have a look at this in a few days, I'm not sure which work-around I prefer, yet, but I'll dive into this at the weekend probably..

**Comment by peti** [comment]
Just for those users who need a fix real quick, add

    kernelParams = ["systemd.legacy_systemd_cgroup_controller=yes"];

to `configuration.nix`. This is not a good solution, but it works for the time being.

**Comment by Mic92** [comment]
@peti should we probably enable this by default until other consumer of cgroups has adapted to cgroupv2?

**Comment by vcunat** [comment]
Longer-time suggestions upstream: https://github.com/systemd/systemd/pull/4670 but that seems unrealistic for 17.03.

**Comment by globin** [comment]
Fixed pushed to staging in 31ff2ac057
